applications/conceptual-model-editor/src/action/add-semantic-entity-to-visual-model-action.ts

```diff
@@ -0,0 +1,48 @@
+import { WritableVisualModel } from "@dataspecer/core-v2/visual-model";
+import { ModelGraphContextType } from "../context/model-context";
+import { UseNotificationServiceWriterType } from "../notification/notification-service-context";
+import { UseDiagramType } from "../diagram/diagram-hook";
+import { sourceModelOfEntity } from "../util/model-utils";
+import { isSemanticModelClass, isSemanticModelGeneralization, isSemanticModelRelationship } from "@dataspecer/core-v2/semantic-model/concepts";
+import { isSemanticModelClassUsage, isSemanticModelRelationshipUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { addSemanticClassToVisualModelAction } from "./add-class-to-visual-model";
+import { addSemanticClassProfileToVisualModelAction } from "./add-class-profile-to-visual-model";
+import { addSemanticRelationshipToVisualModelAction } from "./add-relationship-to-visual-model";
+import { addSemanticRelationshipProfileToVisualModelAction } from "./add-relationship-profile-to-visual-model";
+import { addSemanticGeneralizationToVisualModelAction } from "./add-generalization-to-visual-model";
+
+export function addSemanticEntityToVisualModelAction(
```

Comment:
You are right it is kind of a lazy method. I feel like the only relevant use-case is when you want to add selection to canvas. 

The question is if it will be even implemented in future. I mean I implemented the systematic selection that it can work on top of not visible entities, so in that sense the support is there, but the UI support is not there and I am not sure if there will be.

Reply:
Actually nevermind, it will be used for the use-case when we want to put whole semantic model on canvas. So there is at least one use-case

Reply:
I still can remove the method and implement it without it (it will be actually a bit more performant since we know what entities are nodes and what edges), if you think it seduces to laziness.